File: programmers/문자열압축.py
def solution(s):
    answer = 1e9
    for k in range(1, len(s)+1):
        l = []
        for i in range(0, len(s), k):
            l.append(s[i:i+k])
        cur = l[0]
        cnt = 1
        res = []
        for i in range(1, len(l)):
            if l[i] == cur:
                cnt += 1
            else:
                if cnt > 1:
                    res.append(str(cnt))
                res.append(cur)
                cur = l[i]
                cnt = 1
        res.append(str(cnt) + cur if cnt > 1 else cur)
        answer = min(answer, len(''.join(res)))
    return answer

File: programmers/문자열압축.java
import java.util.*;

class Solution {
    public int solution(String s) {
        int answer = Integer.MAX_VALUE;
        for (int k=1; k<=s.length(); k++) {
            List<String> l = new ArrayList<>();
            for (int i=0; i<s.length()-k+1; i+=k) {
                l.add(s.substring(i, i+k));
            }
            if (s.length()%k >= 1) {
                l.add(s.substring(s.length()-s.length()%k, s.length()));
            }
            
            String cur = l.get(0);
            int cnt = 1;
            StringBuilder sb = new StringBuilder();
            for (int i=1; i<l.size(); i++) {
                if (l.get(i).equals(cur)) {
                    cnt++;
                } else {
                    if (cnt > 1) {
                        sb.append(String.valueOf(cnt));
                    }
                    sb.append(cur);
                    cur = l.get(i);
                    cnt = 1;
                }
            }
            sb.append(cnt > 1 ? String.valueOf(cnt)+cur : cur);
            answer = Math.min(answer, sb.toString().length());
        }
        return answer;
    }
}
